services/storage/src/simcore_service_storage/modules/celery/backends/_redis.py

```diff
@@ -26,26 +33,64 @@ class RedisTaskInfoStore:
     def __init__(self, redis_client_sdk: RedisClientSDK) -> None:
         self._redis_client_sdk = redis_client_sdk
 
-    async def exists(self, task_id: TaskID) -> bool:
+    async def create_task(
+        self,
+        task_id: TaskID,
+        task_metadata: TaskMetadata,
+        expiry: timedelta,
+    ) -> None:
+        task_key = _build_key(task_id)
+        await self._redis_client_sdk.redis.hset(
+            name=task_key,
+            key=_CELERY_TASK_METADATA_KEY,
+            value=task_metadata.model_dump_json(),
+        )  # type: ignore
+        await self._redis_client_sdk.redis.expire(
+            task_key,
+            expiry,
+        )
+
+    async def exists_task(self, task_id: TaskID) -> bool:
         n = await self._redis_client_sdk.redis.exists(_build_key(task_id))
         assert isinstance(n, int)  # nosec
         return n > 0
 
-    async def get_metadata(self, task_id: TaskID) -> TaskMetadata | None:
-        result = await self._redis_client_sdk.redis.hget(_build_key(task_id), _CELERY_TASK_METADATA_KEY)  # type: ignore
-        return TaskMetadata.model_validate_json(result) if result else None
+    async def get_task_metadata(self, task_id: TaskID) -> TaskMetadata | None:
+        raw_result = await self._redis_client_sdk.redis.hget(_build_key(task_id), _CELERY_TASK_METADATA_KEY)  # type: ignore
+        if not raw_result:
+            return None
+
+        try:
+            return TaskMetadata.model_validate_json(raw_result)
+        except ValidationError as exc:
+            _logger.debug(
+                "Failed to deserialize task metadata for task %s: %s", task_id, f"{exc}"
+            )
+            return None
 
-    async def get_progress(self, task_id: TaskID) -> ProgressReport | None:
-        result = await self._redis_client_sdk.redis.hget(_build_key(task_id), _CELERY_TASK_PROGRESS_KEY)  # type: ignore
-        return ProgressReport.model_validate_json(result) if result else None
+    async def get_task_progress(self, task_id: TaskID) -> ProgressReport | None:
+        raw_result = await self._redis_client_sdk.redis.hget(_build_key(task_id), _CELERY_TASK_PROGRESS_KEY)  # type: ignore
+        if not raw_result:
+            return None
 
-    async def get_uuids(self, task_context: TaskContext) -> set[TaskUUID]:
+        try:
+            return ProgressReport.model_validate_json(raw_result)
+        except ValidationError as exc:
+            _logger.debug(
+                "Failed to deserialize task progress for task %s: %s", task_id, f"{exc}"
+            )
+            return None
+
+    async def list_tasks(self, task_context: TaskContext) -> list[Task]:
         search_key = (
             _CELERY_TASK_INFO_PREFIX
             + build_task_id_prefix(task_context)
             + _CELERY_TASK_ID_KEY_SEPARATOR
         )
-        keys = set()
+        search_key_len = len(search_key)
+
+        keys: list[str] = []
+        pipe = self._redis_client_sdk.redis.pipeline()
```

Comment:
pipe and pipelines are differente things, I would use pipeline there, for me pipe is confusing
```suggestion
        pipeline = self._redis_client_sdk.redis.pipeline()
```